pkg/genericapiserver/config.go

```diff
@@ -399,7 +399,9 @@ func (c *Config) ApplyDelegatingAuthenticationOptions(o *options.DelegatingAuthe
 	}
 
 	c.Authenticator = authenticator
-	c.OpenAPIConfig.SecurityDefinitions = securityDefinitions
+	if c.OpenAPIConfig != nil {
```

Comment:
This means that we have to call `ApplyDelegatingAuthenticationOptions` **after** `ApplySecureServingOptions`. Same is true for `applyClientCert`. This asks for trouble. Here it's even security related.